### PR TITLE
Dala 3741 enable single request for non premium users

### PR DESCRIPTION
**Pull request by marcelrasbachdfine**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [ ] A peer-review has been executed
  - [ ] The code has been manually inspected by someone who did not implement the feature
- [ ] The PR actually implements what is described in the JIRA-Issue
- [ ] At least one test exists testing the new feature
  - [ ] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [ ] Documentation is updated as required
- [ ] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [ ] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [ ] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [ ] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server
- [ ] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-frontend/src/components/pages/SingleDataRequest.vue [hunk 1]
```diff
@@ -176,16 +249,30 @@ export default defineComponent({
...
       singleDataRequestModel: {},
       footerContent,
       fetchedCompanyInformation: {} as CompanyInformation,
       frameworkOptions: [] as { value: DataTypeEnum; label: string }[],
       frameworkName: this.$route.query.preSelectedFramework as DataTypeEnum,
       contactsAsString: "",
-      dataRequesterMessage: "",
+      allowAccessDataRequesterMessage: false,
+      dataRequesterMessage: "Please provide a valid email before entering a message",
+      dataRequesterMessageNotAllowedText: "Please provide a valid email before entering a message",
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/src/components/pages/SingleDataRequest.vue:270]: Warum gibt es diesen String doppelt?

File: dataland-frontend/src/components/pages/SingleDataRequest.vue [hunk 2]
```diff
@@ -213,6 +302,109 @@ export default defineComponent({
     },
   },
   methods: {
+    /**
+     * Returns the constant
+     * @class
+     */
+    MAX_NUMBER_OF_DATA_REQUESTS_PER_DAY_FOR_ROLE_USER() {
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/src/components/pages/SingleDataRequest.vue:309]: Warum kann man die Konstante nicht direkt verwenden? Das fühlt sich irgendwie überflüssig an.

File: dataland-frontend/src/components/pages/SingleDataRequest.vue [hunk 3]
```diff
@@ -213,6 +302,109 @@ export default defineComponent({
...
+    closeMaxRequestsReachedModal() {
+      this.maxRequestReachedModalIsVisible = false;
+    },
+    /**
+     * Checks if the first email in a string of comma separated emails is valid
+     * @returns true if valid, false otherwise
+     */
+    areValidContacts(): boolean {
+      let contactsValid = false;
+      for (const selectedContact of this.selectedContacts) {
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/src/components/pages/SingleDataRequest.vue:336]: Diese Funktion würde ich etwas kompakter schreiben.

File: dataland-frontend/src/components/pages/SingleDataRequest.vue [hunk 4]
```diff
@@ -213,6 +302,109 @@ export default defineComponent({
...
+      }
+      return contactsValid;
+    },
+
+    /**
+     * Checks if an email string is a valid email by checking for _@_._
+     * @param email the email string to check
+     * @returns true if the email is valid, false otherwise
+     */
+    isValidEmail(email: string): boolean {
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/src/components/pages/SingleDataRequest.vue:348]: Diese Funktion sollte einen RegEx chack machen.

File: dataland-frontend/src/components/pages/SingleDataRequest.vue [hunk 5]
```diff
@@ -213,6 +302,109 @@ export default defineComponent({
...
+          this.dataRequesterMessage = this.dataRequesterMessageNotAllowedText;
+        }
+      }
+    },
+
+    /**
+     * Updates if the message terms and conditions not being accepted should stop the user from submitting the request.
+     * Based on if they are accepted or not and on if the user wants to submit a message
+     */
+    updateConditionsNotAcceptedError(): void {
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/src/components/pages/SingleDataRequest.vue:387]: Warum sind die beiden update Funktionen nicht eher computed properties?

File: dataland-community-manager/src/test/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestManagerTest.kt [hunk 6]
```diff
@@ -124,6 +149,31 @@ class SingleDataRequestManagerTest {
         testWhichEmailMessageIsSentFor(setOf(), null, 1, 0)
     }
 
+    @Test
+    fun `send single data requests as non premium user and verify that the quota is met`() {
+        for (i in 1..maxRequestsForUser) {
+            val passedRequest = sampleRequest.copy(reportingPeriods = setOf(i.toString()))
```
Comment (SiegfriedSobkowiak) [inline dataland-community-manager/src/test/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestManagerTest.kt:155]: Die Zeile ist mir nicht ganz klar. Sollten wir kurz besprechen.

File: dataland-community-manager/src/test/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestManagerTest.kt [hunk 7]
```diff
@@ -157,4 +207,15 @@ class SingleDataRequestManagerTest {
                 anyString(),
             )
     }
+
+    private fun mockSecurityContext() {
```
Comment (SiegfriedSobkowiak) [inline dataland-community-manager/src/test/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestManagerTest.kt:211]: Braucht man das nicht auch an anderen Stellen für Tests? Also sprich sollte das nicht eher eine utils Funktion sein?

File: dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/api/RequestApi.kt [hunk 8]
```diff
@@ -122,7 +122,7 @@ interface RequestApi {
         consumes = ["application/json"],
         produces = ["application/json"],
     )
-    @PreAuthorize("hasRole('ROLE_PREMIUM_USER')")
+    @PreAuthorize("hasRole('ROLE_PREMIUM_USER') or hasRole('ROLE_USER')")
```
[the remark this answers is not in the document]
Comment (brunoneumannsaavedra) [reply dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/api/RequestApi.kt]: Done.

**Pull request merged**